APE_specutil.rst

```diff
@@ -0,0 +1,481 @@
+frequency, energy, or velocity).  To ease use, the Spectrum1D object will have
+``energy``, ``wavelength``, and ``frequency`` properties that convert the
+`spectral_axis` to the appropriate units.  These are to be considered the
+“center” of the spectral bins.  Bin edges will also be accessible, but the
+exact way of accessing bin_edges will be determined at a later date (as it is
+intertwined with the ``.wcs`` representation, discussed further below).
+``flux`` will be a Quantity, so the units of the spectrum should be accessed as
+``.flux.unit``.  The ``uncertainty``, when provided, will be a Quantity-like
+object with the same shape as the ``flux``.  If it has an `uncertainty_type`
+attribute, `arithmetic operations`_ will propagate uncertainties following
+standard rules (see `the nddata APE for details about the uncertainty attribute
+<https://github.com/astropy/astropy-APEs/blob/master/APE7.rst#proposal-for-an-nddatabase-abstract-base-class>`_).
+
+Dimensionality
+^^^^^^^^^^^^^^
+
+``flux`` and ``uncertainty`` may be multidimensional as long as the last
+dimension matches the shape of ``spectral_axis`` This is meant for fast
+operations on collections of spectra that share the same spectral_axis.  While
+it may seem to conflict with the “1D” in the class name, this name scheme is
+meant to communicate the presence of a single common spectral axis.  For more
+on this, see the section on data cubes below.
+
+Metadata
+^^^^^^^^
+
+Note that the ``meta`` attribute is meant for metadata, e.g., information from
+a FITS header.  A ``Spectrum1D`` object may include additional attributes
+beyond the metadata. In many cases this is best implemented via *subclasses* of
+``Spectrum1D``, but these attributes can also be applied directly by analysis
+steps discussed in the “Analysis tools to be included in specutils” section
+below.  For example, a background subtraction tool would add an attribute
+``background`` to the result spectrum, which is itself a ``Spectrum1D`` with
+the same shape as this one which contains the subtracted background.
+
+Creating Spectrum1D objects
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+While the above properties specify the interface for accessing a `Spectrum1D`,
+these are *not* all required for initialization. At the minimum, the user will
+have to provide a ``flux`` (which can be a Quantity or an array) and the
+``spectral_axis``. The Spectrum1D’s spectral axis can be instantiated with
+either a Quantity object, array, or a pixel-to-spectral transformation function
+(the `wcs` object).
+
+In addition to this information, `specutils` will provide a method for reading
+and writing formats regularly used for spectroscopy.  Common formats include
+IRAF, SDSS, MIDAS, FITS tables, and ascii tables.  This I/O Reading and writing
+a `Spectrum1D` object will use the astropy I/O registry.  A
+`Spectrum1D.create_from_array` method should be available to directly create a
```

Comment:
This is interesting -- this seems to suggest that `Spectrum1D` objects with their `spectral_axis` pre-defined as an array cannot be instantiated in the same way a `Spectrum1D` with a wcs object is instantiated?

```python
spec = Spectrum1D(data, wcs=my_wcs)
```

vs.

```python
spec = Spectrum1D.create_from_array(data, spectral_axis=wavelengths)
```

Are we explicitly not supporting spectrum creation via the object's call method for the latter case?

Reply:
(Saw the examples -- perhaps I'm missing something here?)

Reply:
Actually, yes, I would suggest removing create_from_array.  That is more an implementation detail vs the actual functionality that we want (which would be what is in the examples)